pkg/commands/resolver.go

```diff
@@ -252,12 +277,49 @@ func resolveFile(f string, builder build.Interface, pub publish.Interface, so *o
 		return nil, err
 	}
 
-	if so.Selector != "" {
-		b, err = resolve.FilterBySelector(b, so.Selector)
-		if err != nil {
+	var docNodes []*yaml.Node
+
+	// The loop is to support multi-document yaml files.
+	// This is handled by using a yaml.Decoder and reading objects until io.EOF, see:
+	// https://github.com/go-yaml/yaml/blob/v2.2.1/yaml.go#L124
```

Comment:
This seems outdated if we moved to v3?

Reply:
still necessary - the link can be updated